workers/tests/integrationtests/test_api_report.py

```diff
@@ -5,159 +5,120 @@
 # API request tests are seperated for indicator and report
 # because of a bug when using two schemata.
 
-import os
 import unittest
+from typing import Optional
 
-import geojson
 from fastapi.testclient import TestClient
+from schema import Schema
 
 from ohsome_quality_analyst.api import app
 from ohsome_quality_analyst.reports.simple_report.report import SimpleReport
 
-from .utils import api_schema_report, oqt_vcr
+from .api_response_schema import get_feature_schema, get_response_schema
+from .utils import oqt_vcr
 
 
 class TestApiReport(unittest.TestCase):
     def setUp(self):
         self.client = TestClient(app)
 
+        self.report_name = "SimpleReport"
         # Heidelberg
         self.dataset = "regions"
         self.feature_id = 3
         self.fid_field = "ogc_fid"
 
-        # Heidelberg Altstadt
-        # Choose a small enough region to not trigger area size limit
-        infile = os.path.join(
-            os.path.dirname(os.path.abspath(__file__)),
-            "fixtures",
-            "heidelberg-altstadt-feature.geojson",
-        )
-        with open(infile, "r") as f:
-            self.bpolys = geojson.load(f)
-
-        self.report_name = "SimpleReport"
-        self.schema = api_schema_report
+        self.response_schema = get_response_schema()
+        self.feature_schema = get_feature_schema()
 
-    @oqt_vcr.use_cassette()
-    def test_get_report_bpolys(self):
-        url = "/report/{0}?bpolys={1}".format(self.report_name, self.bpolys)
-        response = self.client.get(url)
+        report = SimpleReport()
+        report.set_indicator_layer()
+        self.number_of_indicators = len(report.indicator_layer)
 
+    def run_tests(self, response) -> None:
         self.assertEqual(response.status_code, 200)
-
-        report = response.json()
-        self.schema.validate(report)  # Print information if validation fails
-        self.assertTrue(self.schema.is_valid(report))
+        response = response.json()
```

Comment:
That's not very pretty to overwrite the given parameter. Can be sometimes problematic even though in this case not problematic. I would use `json_response` or something else that comes handy to you.

Reply:
Yes, true.. I renamed it to response_json.